sos/component.py

```diff
@@ -384,14 +388,40 @@ def _setup_logging(self):
             ui_fhandler.setFormatter(logging.Formatter(
                 '%(asctime)s %(levelname)s: %(message)s'))
 
-            self.ui_log.addHandler(ui_fhandler)
+            ui_fhandler.name = 'ui_fhandler'
+            self.add_handler(self.ui_log, ui_fhandler)
 
         if not self.opts.quiet:
             self.add_ui_log_to_stdout()
 
     def get_temp_file(self):
         return self.tempfile_util.new()
 
+    def handler_present(self, name):
+        result = False
+        loggers = []
+        if self.soslog is not None:
+            loggers.append(self.soslog)
+        if self.ui_log is not None:
+            loggers.append(self.ui_log)
+        for logger in loggers:
+            for handler in logger.handlers:
+                if handler.name == name:
+                    result = True
+                    break
+        return result
```

Comment:
This method seems redundant for what is going on in `add_handler()` in the `else` condition.

Since we're already passing the logger to `add_handler()`, we should just inspect that logger directly.